### PR TITLE
HI Pixel Tracking: change fitter from Conformal Mapping to Broken Line/Riemann

### PR DESCRIPTION
**Pull request by CesarBernardes**
Dear All, 

We change configuration files for heavy-ions pixel tracking collection in MiniAOD to use BrokenLine/Riemman fitters.

We re-tune loose pre-selection and lower pT threshold from 0.3GeV/c to 0.2GeV/c. Since 
new fitters have much higher absolute efficiency in the pT range of interest of this collection, i.e. ~0.2-1.0GeV/c (please, see slides 2-3 of [*]), the total size of MiniAOD heavy-ion event is expected to increase by around 2-3% (with small dependency on collision centrality). Please, see slide 5 of [*].

We did timing studies. With new configuration we have almost factor 3 of reduction in timing for pixel tracking 
collection sequence. Please, see slide 4 of [*].

[*] https://indico.cern.ch/event/1107099/contributions/4658135/attachments/2366348/4040751/slides_HINTRKmeeting_16122021.pdf

Best Regards,

   Cesar A. Bernardes

@abaty @mandrenguyen

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: PhysicsTools/PatAlgos/python/slimming/hiPixelTracks_cfi.py [hunk 1]
```diff
@@ -3,6 +3,11 @@
 from PhysicsTools.PatAlgos.patTracksToPackedCandidates_cfi import patTracksToPackedCandidates
 
 hiPixelTracks = patTracksToPackedCandidates.clone()
+hiPixelTracks.dxySigCut = 3.
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/slimming/hiPixelTracks_cfi.py:6]: please update parameters during cloning
```suggestion
hiPixelTracks = patTracksToPackedCandidates.clone(
  dxySigCut = 3.
)
```
(similarly for the other parameters)

Comment (CesarBernardes) [reply PhysicsTools/PatAlgos/python/slimming/hiPixelTracks_cfi.py]: Thanks @slava77 , I fixed this doing this commit https://github.com/cms-sw/cmssw/pull/36558/commits/f3d9ca78f2748d6108378188f1a7b6c80ed2d5dd

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
The size test was done with 20 events if I read the slides correctly (Central collisions b=4fm (20 events). Could you confirm if the 2-3% holds also for larger stats (e.g. 10x more)?
Thanks.

**Comment by CesarBernardes** [comment]
> The size test was done with 20 events if I read the slides correctly (Central collisions b=4fm (20 events)). Could you confirm if the 2-3% holds also for larger stats (e.g. 10x more)? Thanks.

Hi @jpata , Thanks for the suggestion. We did a test now using 200 events for b=4fm (they are indeed very central PbPb collisions, centrality~<3%), and we got 966kB/event for new configuration, while for default we got 943kB/event. That corresponds to around 2.4% increase in size for new configuration, I would say is compatible with the studies using 20 events.

**Comment by jpata** [comment]
+reconstruction
- new fitter for HI tracking, along with new quality cuts (significantly improved signal efficiency has been reported)
- reco changes in HI tracks expected/observed
- HI MINIAOD size increases by a few %

**Comment by jpata** [comment]
@CesarBernardes please fix the typos in the PR title!

**Comment by mandrenguyen** [comment]
@CesarBernardes   The change of event size you quote is for MINIAOD and not MINIAODSiM, correct?  What is the change in size to the AOD?

**Comment by CesarBernardes** [comment]
> @CesarBernardes please fix the typos in the PR title!

@jpata, sorry, I fixed this now.

**Comment by CesarBernardes** [comment]
> @CesarBernardes The change of event size you quote is for MINIAOD and not MINIAODSiM, correct? What is the change in size to the AOD?

Hi @mandrenguyen, I always had in mind that was MINIAOD, but I cross-checked after your question and I see in configuration that is MINIAODSIM. Should we recheck for MINIAOD only? I did not check AOD, I can do this also.

**Comment by mandrenguyen** [comment]
Yes, please check for miniAOD and AOD.  As you know, it’s of utmost  importance to decrease the data volume, for which tracks are one of the biggest consumers.  A few percent increase is not at all negligible, and will be larger when you drop the sim collections.  What is the physics case for keeping such low pt tracks?

**Comment by CesarBernardes** [comment]
> Yes, please check for miniAOD and AOD. As you know, it’s of utmost importance to decrease the data volume, for which tracks are one of the biggest consumers. A few percent increase is not at all negligible, and will be larger when you drop the sim collections. What is the physics case for keeping such low pt tracks?
> […](#)
> Sent from my iPhone
> On Jan 7, 2022, at 4:25 PM, Cesar Bernardes ***@***.***> wrote: ﻿ @CesarBernardes The change of event size you quote is for MINIAOD and not MINIAODSiM, correct? What is the change in size to the AOD? Hi @mandrenguyen, I always had in mind that was MINIAOD, but I cross-checked after your question and I see in configuration that is MINIAODSIM. Should we recheck for MINIAOD only? I did not check AOD, I can do this also. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.

Sure, I understand. I will check both MINIAOD and AOD. The physics case is mainly for flow and correlations analyses. We have analyses that are important (good fraction of phase space) to go below pT=0.5GeV/c. Examples are: flow fluctuations from Event-by-Event analysis and short range correlations (like charge balance function and net charge fluctuations,  femtoscopic correlations, etc...).  In fact, when we do these analyses in pp and pPb we go until 0.3GeV/c and would also be important to have same threshold in PbPb, but with general tracks the fake-rate is very big compared to pixel tracking for mid-central to central collisions.

**Comment by mandrenguyen** [comment]
I’m not asking what the physics case for going below 500 MeV.  I’m asking what we gain by going all the way down to 200 MeV, in light of the increase in event size, compared to the 300 MeV cut

**Comment by CesarBernardes** [comment]
> I’m not asking what the physics case for going below 500 MeV. I’m asking what we gain by going all the way down to 200 MeV, in light of the increase in event size, compared to the 300 MeV cut
> […](#)
> Sent from my iPhone
> On Jan 7, 2022, at 5:48 PM, Cesar Bernardes ***@***.***> wrote: ﻿ Yes, please check for miniAOD and AOD. As you know, it’s of utmost importance to decrease the data volume, for which tracks are one of the biggest consumers. A few percent increase is not at all negligible, and will be larger when you drop the sim collections. What is the physics case for keeping such low pt tracks? … Sent from my iPhone On Jan 7, 2022, at 4:25 PM, Cesar Bernardes @.***> wrote: ﻿ @CesarBernardes The change of event size you quote is for MINIAOD and not MINIAODSiM, correct? What is the change in size to the AOD? Hi @mandrenguyen, I always had in mind that was MINIAOD, but I cross-checked after your question and I see in configuration that is MINIAODSIM. Should we recheck for MINIAOD only? I did not check AOD, I can do this also. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned. Sure, I understand. I will check both MINIAOD and AOD. The physics case is mainly for flow and correlations analyses. We have analyses that are important (good fraction of phase space) to go below pT=0.5GeV/c. Examples are: flow fluctuations from Event-by-Event analysis and short range correlations (like charge balance function and net charge fluctuations, femtoscopic correlations, etc...). In fact, when we do these analyses in pp and pPb we go until 0.3GeV/c and would also be important to have same threshold in PbPb, but with general tracks the fake-rate is very big compared to pixel tracking for mid-central to central collisions. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.

sorry, I see the point. In fact, the decision on this pT threshold was mostly based on tracking performance only. I will try a study checking the increase in size also in MINIAOD and AOD. In this case is better I also present the studies in the HIN Tracking group meeting. I will check this for pT>0.20, 0.25, 0.30 GeV/c and will let you know. Thanks.

**Comment by perrotta** [comment]
hold
(@CesarBernardes @mandrenguyen please let us know when the studies announced in https://github.com/cms-sw/cmssw/pull/36558#issuecomment-1007586540 will have reached any conclusion about the pt threshold and the corresponding increase in the HI event size)

**Comment by jpata** [comment]
I had (mistakenly) assumed that this came already with the blessing of the HIN group, thanks Matt for checking/correcting...

**Comment by jpata** [comment]
Note that:
- a size check on AOD
- an approval by HI (following a discussion and links to slides in a HI meeting)
are pending here for the reco sig, just redacted.

**Comment by CesarBernardes** [comment]
> Note that:
> 
>     * a size check on AOD
> 
>     * an approval by HI (following a discussion and links to slides in a HI meeting)
>       are pending here for the reco sig, just redacted.

Hi @jpata , @perrotta , @mandrenguyen , after discussion in the HIN tracking meeting (https://indico.cern.ch/event/1113861/), we decided to use pT range for pixel tracking in MINIAOD of 0.3<pT<1GeV (this is the same pT configuration as used before this PR. I just made a commit for this https://github.com/cms-sw/cmssw/pull/36558/commits/19c31aeae788f12c2cf0eb25655e4f01ba6e8a03). We show studies in slide 3 of (https://indico.cern.ch/event/1113861/contributions/4680588/attachments/2372534/4052666/slides_HINTRKmeeting_13012022.pdf) . Note that the size in MINIAOD increases by around 8% with this configuration, but the size in AOD decreases by around 5% . The AOD size decreases because we apply a tighter track selection during the reconstruction to produce the collection in AOD (chi2 and DCA significance cuts https://github.com/cms-sw/cmssw/pull/36558/commits/b2f7612c7e288a46799dac760e42a728e74a8113#diff-62cf1aed089b48edc60b97d0808a49dc4bda09456580fef24793002c9c468d13).

**Comment by mandrenguyen** [comment]
Just to explicitly sign off (for what it's worth), we agreed to keep the same physics reach (down to 300 MeV), and focus this PR on the considerable improvement on timing and efficiency.  While it's true that this increases the size of miniAOD, due to the increased efficiency, this is more than compensated by the decreased size of AOD, due to the tightened cuts.  Since our datasets are usually not replicated, this is an overall win.  Increasing the reach to 200 MeV is postponed until studies can be carried out in the relevant physics group.  They should also consider decreasing the upper cut for pixel tracks from 1 GeV to 800 MeV.  Apologies for the noise.

**Comment by perrotta** [comment]
unhold
(Thank you @CesarBernardes and @mandrenguyen . Now it's up to reco to evaluate and eventually approve the changes in the output size)

**Comment by perrotta** [comment]
please test

**Comment by jpata** [comment]
assign @cms-sw/xpog-l2

**Comment by qliphy** [comment]
assign xpog

> @cmsbuild assign xpog

**Comment by mariadalfonso** [comment]
+xpog
no objections to the few % increase for miniAOD HI

**Comment by qliphy** [comment]
please test
to refresh

**Comment by perrotta** [comment]
+1
- Event size changes are approved by reco and xpog

**Pull request merged**